command/agent/dns.go

```diff
@@ -33,6 +33,37 @@ type DNSServer struct {
 	logger       *log.Logger
 }
 
+// Shutdown stops the DNS Servers
+func (d *DNSServer) Shutdown() error {
+	// Async stop the DNS Servers, handle a potential error
+	errCh := make(chan error, 1)
+	go func(d *DNSServer) {
+		if err := d.dnsServer.Shutdown(); err != nil {
+			d.logger.Printf("[ERR] dns: error stopping udp server: %v", err)
+			errCh <- fmt.Errorf("dns udp teardown failed: %v", err)
```

Comment:
Since we don't use this error from the caller due to the nature of the shutdown path (no opportunity to recover), the error channels might not really be necessary, and the signature of this function could be modified to return nothing (just log the errors and move on).